### ISSUE TITLE
Error on creating subscription with payment done

### ISSUE THREAD

**Issue by aureleduda** [ptibogxiv/adherentsplus#32]
Since update to dolibarr 18.0.1 and adherentsplus 18.0.0 (same with 18.0.1) I cannot create a subscription with the invoice payed. It returns me a in debug bar :
Undefined index: cancel (/home/mfppxnud/admin.mfpp.fr/adherentsplus/subscription.php:223)
NOTICE Trying to get property 'code' of non-object (/home/mfppxnud/admin.mfpp.fr/compta/bank/class/account.class.php:585
And I get a red mesg "oper not defined"

If I create a subscription with a invoice without payment, it works, but it create an invoice with a amount completely wrong (on my test, 2024€ in place of 45€, seems to be the year of the subscription in place of the amount).
I have check to send an email to the member, but it was send to another member (the lower ID number of the base). The debugBar said the memberid was invalid.

**Comment by aureleduda** [comment]
I made some check with doli debug, and the log show that when I create subscription with invoiceonly : 

```
2023-10-15 06:12:17 DEBUG   176.165.163.166    sql=INSERT INTO tableprefix_subscription (fk_adherent, fk_type, datec, dateadh, datef, subscription, note) VALUES (3165, '2', '2023-10-15 08:12:17', '2024-01-01 00:00:00', '2024-12-31 23:59:59', 30, 'Adhésion / cotisation 2024')
2023-10-15 06:12:17 INFO    176.165.163.166    Execute a trigger MEMBER_SUBSCRIPTION_CREATE with context key 'member' not an object
```
then on invoice preparation : 
```
2023-10-15 06:12:17 DEBUG   176.165.163.166     sql=INSERT INTO tableprefix_actioncomm(ref,datec,datep,datep2,durationp,fk_action,code,ref_ext,fk_soc,fk_project,note,fk_contact,fk_user_author,fk_user_action,fk_user_done,label,percent,priority,fulldayevent,location,transparency,fk_element,elementtype,entity,extraparams,email_msgid,email_from,email_sender,email_to,email_tocc,email_tobcc,email_subject,errors_to,recurid,recurrule,recurdateend,num_vote,event_paid,status,ip) VALUES ('(PROV)', '2023-10-15 08:12:17', '2023-10-15 08:12:17', '2023-10-15 08:12:17', null, 40,'AC_MEMBER_SUBSCRIPTION_CREATE', null, null, null,  'Cotisation 900 pour l\'adhérent NAME ajoutée\nAdhérent: NAME\nType: 2\nMontant: 30\nPériode: 01/01/2024 - 31/12/2024', null, 1, 1, null, 'Cotisation 900 pour l\'adhérent NAME ajoutée', '-1', '0', '0', '', '0', 3165, 'member', 1,null, null, null, null, null, null, null, null, null, null, null, null, null, 0, 0, null)
```
And
```
2023-10-15 06:12:17 DEBUG   176.165.163.166     FactureFournisseur::create
2023-10-15 06:12:17 DEBUG   176.165.163.166     sql=INSERT INTO tableprefix_facture_fourn (ref, ref_supplier, ref_ext, entity, type, libelle, fk_soc, datec, datef, vat_reverse_charge, fk_projet, fk_cond_reglement, fk_mode_reglement, fk_account, note_private, note_public, fk_user_author, date_lim_reglement, fk_incoterms, location_incoterms, fk_multicurrency, multicurrency_code, multicurrency_tx, fk_facture_source, fk_fac_rec_source) VALUES ('(PROV)', '', '', 1, '0', '', 0, '2023-10-15 08:12:17', '2023-10-15 08:12:17', 0, null, null, null, NULL, '', '', 1,null, 0, '', 0, 'EUR', 1, null, NULL)
2023-10-15 06:12:17 ERR     176.165.163.166     DoliDBMysqli::query SQL Error message: DB_ERROR_1452 Cannot add or update a child row: a foreign key constraint fails (`database`.`tableprefix_facture_fourn`, CONSTRAINT `fk_facture_fourn_fk_soc` FOREIGN KEY (`fk_soc`) REFERENCES `tableprefix_societe` (`rowid`))
2023-10-15 06:12:17 DEBUG   176.165.163.166    Adherent::update_end_date
2023-10-15 06:12:17 DEBUG   176.165.163.166    sql=SELECT rowid, datec as dateop, dateadh as datedeb, datef as datefin FROM tableprefix_subscription WHERE fk_adherent = 3165 ORDER by dateadh DESC
2023-10-15 06:12:17 DEBUG   176.165.163.166    Adherent::update_end_date
2023-10-15 06:12:17 DEBUG   176.165.163.166    sql=UPDATE tableprefix_adherent SET datefin='2024-12-31 23:59:59' WHERE rowid = 3165
2023-10-15 06:12:18 INFO    176.165.163.166  subscriptionComplementaryActions subscriptionid=900 option=invoiceonly accountid=-1 datesubscription=1704063600 paymentdate=1735685999 label= amount=Adhésion / cotisation 2024 num_chq=30 autocreatethirdparty=
```
the fk_soc should be 70 in this example but is not passed in the sql (0 instead). 
In subscriptionComplementaryActions, the label is place in amount value, and the num_chq value is the amount. 
Don't know why, but I hope it helps to determine if the problemis from your module.

**Comment by ptibogxiv** [comment]
fix in PR pushed and for futur 18.0.2 version
